src/Cli/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -50,9 +50,27 @@ public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSett
                             .EnumerateObject()
                             .FirstOrDefault(IsDefaultProfileType).Value;
                     }
-                    else
+                    else // Find a profile match for the given profileName
                     {
-                        if (!profilesObject.TryGetProperty(profileName, out profileObject) || profileObject.ValueKind != JsonValueKind.Object)
+                        IEnumerable<JsonProperty> caseInsensitiveProfileMatches = profilesObject
+                            .EnumerateObject() // p.Name shouldn't fail, as profileObject enumerables here are only created from an existing JsonObject
+                            .Where(p => string.Compare(p.Name, profileName, StringComparison.OrdinalIgnoreCase) == 0);
+
+                        if (caseInsensitiveProfileMatches.Count() > 1)
```

Comment:
This could be converted into a switch statement, but that doesn't seem too valuable.